assets/js/blocks/cart-checkout/mini-cart/component-frontend.tsx

```diff
@@ -13,25 +15,114 @@ import {
  * Internal dependencies
  */
 import CartLineItemsTable from '../cart/full-cart/cart-line-items-table';
+import './style.scss';
 
-const MiniCartContents = () => {
-	const { cartItems, cartIsLoading } = useStoreCart();
+interface MiniCartBlock {
+	isPlaceholderOpen?: boolean;
+}
 
-	if ( cartItems.length === 0 ) {
-		return <>{ __( 'Cart is empty', 'woo-gutenberg-products-block' ) }</>;
-	}
+const MiniCartBlock = ( { isPlaceholderOpen = false } ): JSX.Element => {
+	const { cartItems, cartItemsCount, cartIsLoading } = useStoreCart();
+	const [ isOpen, setIsOpen ] = useState< boolean >( isPlaceholderOpen );
+	// We already rendered the HTML drawer placeholder, so we want to skip the
+	// slide in animation.
+	const [ skipSlideIn, setSkipSlideIn ] = useState< boolean >(
+		isPlaceholderOpen
+	);
+
+	const contents =
+		cartItems.length === 0 ? (
+			<>{ __( 'Cart is empty', 'woo-gutenberg-products-block' ) }</>
+		) : (
+			<div className="is-mobile">
+				<CartLineItemsTable
+					lineItems={ cartItems }
+					isLoading={ cartIsLoading }
+				/>
+			</div>
+		);
 
 	return (
-		<CartLineItemsTable
-			lineItems={ cartItems }
-			isLoading={ cartIsLoading }
-		/>
+		<>
+			<button
+				className="wc-block-mini-cart__button"
+				onClick={ () => {
```

Comment:
I wasn't able to reproduce this issue exactly. But looking at the code I think we can safely ignore clicks when the Drawer is already open. I implemented the changes in e65ce2fa7a7ab1cce51a1933450bd7480fb60dac. Can you check if they help? Otherwise, how would you apply debounce here while making the Drawer to open as fast as possible when clicking on the button?

Reply:
I think this was the way to go, you make a good point with the debouncing and loosing the performance of a fast opening Drawer.

Regarding the bug I've mentioned here is [a video I made](https://www.loom.com/share/7818572367364648b2ffa250a44f5ebb) of it. I think this is an edge case, so if you like to take a look at it in another ticket feel free. I am gonna approve this PR.